format.js

```diff
@@ -0,0 +1,111 @@
+const util = require('util');
+
+/**
+ * Pads the provided number into a string of 2 digits; adding leading
+ * zeros if needed.
+ * Values resulting in more than 2 digits are left untouched.
+ *
+ * @param {string} number
+ * @returns A two-digit number
+ * @throws {TypeError} if the provided value is not a valid integer
+ */
+function formatTwoDigits(number) {
+  if (!Number.isInteger(number)) {
+    throw new TypeError(`Not a valid integer: ${number}`);
+  }
+
+  const n = String(number);
+
+  return n.padStart(2, '0');
+}
+
+/**
+ * Returns a [short](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#month)
+ * (3-letters) representation of the given month index **in English**.
+ * @param {number} month
+ * @returns {string}
+ * @throws {TypeError} if the provided value is not a valid month number
+ */
+function formatShortMonth(month) {
+  switch (month) {
+    // Note: January is month 0!
+    case 0:
+      return 'Jan';
+    case 1:
+      return 'Feb';
+    case 2:
+      return 'Mar';
+    case 3:
+      return 'Apr';
+    case 4:
+      return 'May';
+    case 5:
+      return 'Jun';
+    case 6:
+      return 'Jul';
+    case 7:
+      return 'Aug';
+    case 8:
+      return 'Sep';
+    case 9:
+      return 'Oct';
+    case 10:
+      return 'Nov';
+    case 11:
+      return 'Dec';
+    default:
+      throw new TypeError(`Not a valid month value: ${month}`);
+  }
+}
```

Comment:
Not really sure what's the benefit of using a map object here; I'd even argue its an unnecesary memory sink.

This is a logging package; meaning on high traffic services this function will be called **a lot**. We are creating an object, referencing it on memory, consuming 1/12th of its content and then throwing it away. For **every log message**. That's quite some work for the gc just for logging.

We could consider moving the map outside of the function and keep it in memory; but I'd still advise against it. We are using a map object for mapping 12 well-known values with a non-trivial impact on code readability.

Reply:
Went ahead and created a lookup-table for the monts **outside of the function** (to avoid creating short-lived objects) 38c09d29081380b5c2cfb5734123b69db574111f

Reply:
I am pushing for better readability! and I see the memory overhead from creating a constant object registry/mapper is minimal (very low). JavaScript engines can optimize for this. The only missing part that we can refine here is moving the object outside the function.